### PR TITLE
Create world

### PR DESCRIPTION
**Pull request by daniel-deboeverie-lemon**
This merge builds on from https://github.com/olexale/bdd_widget_test/pull/60 since it needed a few changes made there.

This PR adds/fixes the following;
- A addWorld flag to the generator options
- When this flag is true it adds a World object to the generated feature files, step files and hook files
- The World contains a parameters prop that is a key value map that can be edited by the developer.
- Since the hook file is now dynamic, changed it to use the constant declared method names
- Fixed an indent issue for if only hooks were enabled (without an after in the feature files)
- Fixed a formatting issue for the beforeEach hook

### CODE REVIEW

**Comment by lsaudon** [comment]
I'm not familiar with the notion of world in BDD.
Do you have any documentation about it?

**Comment by daniel-deboeverie-lemon** [comment]
https://github.com/cucumber/cucumber-js/blob/main/docs/support_files/world.md

File: README.md [hunk 1]
```diff
@@ -426,6 +426,47 @@ targets:
...
+      bdd_widget_test|featureBuilder:
+        options:
+          addWorld: true
+```
+This will add a world object to the generated feature files, step files and hook files. Do note that it will not regenerate the step files you already have, so you might need to add the world parameter there manually.
+The world parameter holds a key value map that you can store state in that travels through the scenario. This means you could translate the following feature file in these steps:
+```gherkin
+ Scenario: Input login screen
+        Given I have a user {'Tom'} {'password'}
+        When When I enter {'Tom'} into {1} input field
```
Comment (antonderevyanko) [inline README.md:445]: Is this duplication a mistake?

**Review by antonderevyanko — COMMENTED** [review]
A minor suggestion for Readme file

File: README.md (hunk 1 above)
Comment (olexale) [reply README.md:445]: Definitely a typo.

**Comment by olexale** [comment]
As discussed in #62, this might be achieved with good old DI, so I'm closing the PR. @daniel-deboeverie-lemon, thanks once again for your work!

If you believe it would still be beneficial to introduce the World, feel free to reopen the PR, and let's discuss.

**Pull request closed without merging**